kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/StatsSchemaHelper.java

```diff
@@ -0,0 +1,258 @@
+package io.delta.kernel.internal.skipping;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import io.delta.kernel.expressions.Column;
+import io.delta.kernel.expressions.Literal;
+import io.delta.kernel.internal.util.Tuple2;
+import io.delta.kernel.types.*;
+import static io.delta.kernel.internal.util.ColumnMapping.getPhysicalName;
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+
+/**
+ * Provides information and utilities for statistics columns given a table schema. Specifically,
+ * it is used to:
+ * <ol>
+ *     <li>Get the expected statistics schema given a table schema</li>
+ *     <li>Check if a {@link Literal} or {@link Column} is skipping eligible</li>
+ *     <li>Get the statistics column for a given stat type and logical column</li>
+ * </ol>
+ */
+public class StatsSchemaHelper {
```

Comment:
is this all new logic, or copied from some existing logic in delta-spark?

Reply:
Mostly new